### PR TITLE
add zoekt-mirror-gitea

### PR DESCRIPTION
**Pull request by techknowlogick**

### CODE REVIEW

File: cmd/zoekt-mirror-gitea/main.go [hunk 1]
```diff
@@ -0,0 +1,315 @@
...
+	}
+	if *giteaURL == "" && *org == "" && *user == "" {
+		log.Fatal("must set either --org or --user when gitea.com is used as host")
+	}
+
+	var host string
+	var client *gitea.Client
+	clientOptions := []gitea.ClientOption{}
+	if *giteaURL != "" {
+		*giteaURL = "https://gitea.com/"
```
Comment (keegancsmith) [inline cmd/zoekt-mirror-gitea/main.go]: shouldn't you just make the default for the flag this?

File: cmd/zoekt-mirror-gitea/main.go [hunk 2]
```diff
@@ -0,0 +1,315 @@
...
+func filterRepositories(repos []*gitea.Repository, include []string, exclude []string, noArchived bool) (filteredRepos []*gitea.Repository) {
+	for _, repo := range repos {
+		if noArchived && *&repo.Archived {
+			continue
+		}
+		// FIXME: get topics from repos
+		// if (len(include) == 0 || hasIntersection(include, repo.Topics)) &&
+		// 	!hasIntersection(exclude, repo.Topics) {
+		// 	filteredRepos = append(filteredRepos, repo)
+		// }
```
Comment (keegancsmith) [inline cmd/zoekt-mirror-gitea/main.go]: looks like you still need to do this?

File: cmd/zoekt-mirror-gitea/main.go [hunk 3]
```diff
@@ -0,0 +1,315 @@
...
+			"zoekt.archived": marshalBool(r.Archived),
+			"zoekt.fork":     marshalBool(r.Fork),
+			"zoekt.public":   marshalBool(r.Private || r.Internal), // count internal repos as private
+		}
+		dest, err := gitindex.CloneRepo(destDir, r.FullName, r.CloneURL, config)
+		if err != nil {
+			return err
+		}
+		if dest != "" {
+			fmt.Println(dest)
```
Comment (keegancsmith) [inline cmd/zoekt-mirror-gitea/main.go:279]: debug logs?

**Review by keegancsmith — COMMENTED** [review]
looks good so far, just need to finish it up and sign the CLA

File: cmd/zoekt-mirror-gitea/main.go (hunk 2 above)
Comment (techknowlogick) [reply cmd/zoekt-mirror-gitea/main.go]: Whoops. I should've added a better note. The API doesn't have these fields yet, and this is left as an artifact for me in the future when the API is updated.

File: cmd/zoekt-mirror-gitea/main.go (hunk 1 above)
Comment (techknowlogick) [reply cmd/zoekt-mirror-gitea/main.go]: Good catch. Most of this file is copy/paste from the github mirror. I've cleaned this up.

File: cmd/zoekt-mirror-gitea/main.go (hunk 3 above)
Comment (techknowlogick) [reply cmd/zoekt-mirror-gitea/main.go:279]: This is copy/paste from the GitHub mirror code

**Comment by techknowlogick** [comment]
@cla-bot, I have already filled out with email address & github username!

**Review by keegancsmith — APPROVED** [review]
Thanks! I'll check in with others about cla-bot misbehaving.

**Comment by keegancsmith** [comment]
I see your CLA entry, the bot got confused with some trailing whitespace in it. Sorting the bot out. Thanks again for your contribution.

**Pull request merged**